src/plot_settings.py

```diff
@@ -106,6 +107,20 @@ def set_limits(self, parent):
 
     def on_close(self, _, parent):
         plot_settings = parent.plot_settings
+
+        # Check if style change when override is enabled
+        self.style_changed = False
+        if parent.preferences.config["override_style_change"]:
+            parent.canvas = Canvas(parent=parent)
+            # Check if style is changed
+            if plot_settings.use_custom_plot_style != \
+                    self.use_custom_plot_style.get_enable_expansion():
+                self.style_changed = True
+            elif plot_settings.custom_plot_style != \
+                    self.custom_plot_style.get_selected_item().get_string():
+                self.style_changed = True
```

Comment:
Yeah that works as well, the reason for splitting it up was just because it became a bit long and perhaps not more readible. But I think either works. I've changed to a single conditional now as you suggested, as it's a bit cleaner. No need to repeat the self.style_change=True.

Reply:
Why check three times for plot_settings.custom_plot_style though?

Reply:
It's a sign I need to get lunch, not sure why I did it like that. That was obviously not intended. 
Your suggested line works fine actually.